packages/users/src/api/index.ts

```diff
@@ -0,0 +1,323 @@
+export async function sendMessage(
+  api: GrpcConnection,
+  from: string,
+  to: string,
+  toBody: Uint8Array,
+  toSignature: Uint8Array,
+  fromBody: Uint8Array,
+  fromSignature: Uint8Array,
+  ctx?: ContextInterface,
+): Promise<UserMessage> {
+  logger.debug('send message request')
+  const req = new SendMessageRequest()
+  req.setTo(to)
+  req.setTobody(toBody)
+  req.setTosignature(toSignature)
+  req.setFrombody(fromBody)
+  req.setFromsignature(fromSignature)
+  const res: SendMessageReply = await api.unary(API.SendMessage, req, ctx)
+  return {
+    id: res.getId(),
+    createdAt: res.getCreatedat(),
+    body: fromBody,
+    signature: fromSignature,
+    to,
+    from,
+  }
+}
+
+/**
+ * @internal
+ */
+export async function listInboxMessages(
+  api: GrpcConnection,
+  opts?: InboxListOptions,
+  ctx?: ContextInterface,
+): Promise<Array<UserMessage>> {
+  logger.debug('list inbox message request')
+  const req = new ListInboxMessagesRequest()
+  if (opts && opts.seek) req.setSeek(opts.seek)
+  if (opts && opts.limit) req.setLimit(opts.limit)
+  if (opts && opts.ascending) req.setAscending(opts.ascending)
+  if (opts && opts.status) req.setStatus(opts.status)
+  const res: ListMessagesReply = await api.unary(API.ListInboxMessages, req, ctx)
+  return res.getMessagesList().map(convertMessageObj)
+}
+
+/**
+ * @internal
+ */
+export async function listSentboxMessages(
+  api: GrpcConnection,
+  opts?: SentboxListOptions,
+  ctx?: ContextInterface,
+): Promise<Array<UserMessage>> {
+  logger.debug('list sentbox message request')
+  const req = new ListSentboxMessagesRequest()
+  if (opts && opts.seek) req.setSeek(opts.seek)
+  if (opts && opts.limit) req.setLimit(opts.limit)
+  if (opts && opts.ascending) req.setAscending(opts.ascending)
+  const res: ListMessagesReply = await api.unary(API.ListSentboxMessages, req, ctx)
+  return res.getMessagesList().map(convertMessageObj)
+}
+
+/**
+ * @internal
+ */
+export async function readInboxMessage(
+  api: GrpcConnection,
+  id: string,
+  ctx?: ContextInterface,
+): Promise<{ readAt: number }> {
+  logger.debug('read inbox message request')
+  const req = new ReadInboxMessageRequest()
+  req.setId(id)
+  const res: ReadInboxMessageReply = await api.unary(API.ReadInboxMessage, req, ctx)
+  return { readAt: res.toObject().readat }
+}
+
+/**
+ * @internal
+ */
+export async function deleteInboxMessage(api: GrpcConnection, id: string, ctx?: ContextInterface) {
+  logger.debug('delete inbox message request')
+  const req = new DeleteMessageRequest()
+  req.setId(id)
+  await api.unary(API.DeleteInboxMessage, req, ctx)
+  return
+}
+
+/**
+ * @internal
+ */
+export async function deleteSentboxMessage(api: GrpcConnection, id: string, ctx?: ContextInterface) {
+  logger.debug('delete sentbox message request')
+  const req = new DeleteMessageRequest()
+  req.setId(id)
+  await api.unary(API.DeleteSentboxMessage, req, ctx)
+  return
+}
+
+/**
+ * @internal
+ */
+export function watchMailbox(
```

Comment:
new listen method for mailbox @carsonfarmer

Reply:
Sweet, yup looks good. One day we'll get those Buffer.from's out of everything.